drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -5467,14 +5484,14 @@ RenderingDeviceDriverD3D12::RenderingDeviceDriverD3D12(D3D12Context *p_context,
 
 				CD3DX12_RESOURCE_DESC resource_desc = CD3DX12_RESOURCE_DESC::Buffer(D3D12_CONSTANT_BUFFER_DATA_PLACEMENT_ALIGNMENT);
 
-				ID3D12Resource *resource = nullptr;
+				ComPtr<ID3D12Resource> resource;
```

Comment:
This particular line or in general?
This is the correct way to not leak on teardown.
If you search for other usages of D3D12MA::Allocator::CreateResource() in here it's the same.

Reply:
I was asking about this one in particular. That change will indeed avoid leaking, but not for the right reason: if the resource pointers is immediately assigned to the `ComPtr`, when the scope ends, it will be released. A possible approach to avoid the leak while keeping the object alive long enough would be to release every `frames[i].aux_resource` in `~RenderingDeviceDriverD3D12()`.